app/src/main/java/org/kochka/android/weightlogger/tools/GarminConnect.java

```diff
@@ -136,22 +142,32 @@ public boolean signin(final String username, final String password) {
       // Follow redirections
       get = createHttpGetFromLocationHeader(getTicketLocation);
       get.setParams(params);
-      httpclient.execute(get);
-
-      // Initialise session. Redirect manually, as there are two URLs
-      // marked by HttpClient as duplicates (but in fact these differ by queryparams).
-      /*
-      RedirectStrategy oldRedirectStrategy = httpclient.getRedirectStrategy();
-      httpclient.setRedirectStrategy(new NoRedirectStrategy());
-      CloseableHttpResponse initSessionResponse = httpclient.execute(new HttpGet(LEGACY_INIT_SESSION_URL));
-      Header initSessionLocation = initSessionResponse.getFirstHeader("location");
-      for(int i=0; i<5; i++){
-        get = createHttpGetFromLocationHeader(initSessionLocation);
-        get.setParams(params);
-        initSessionLocation = httpclient.execute(get).getFirstHeader("location");
+      HttpEntity entity3 = httpclient.execute(get).getEntity();
+      String entity3ResponseAsString = EntityUtils.toString(entity3);
+
+      if (isSignedIn(username)) {
+
+        // Get oauth v1 token
+        HttpGet getOauth1 = new HttpGet(GET_OAUTH1_URL + ticket);
+        getOauth1.setParams(params);
+        HttpEntity oauth1Entity = httpclient.execute(getOauth1).getEntity();
```

Comment:
This request needs to be signed with the [OAuth1 consumer keys](https://thegarth.s3.amazonaws.com/oauth_consumer.json)

Reply:
Ah this is the missing piece no doubt! Not sure how I scanned over that in your walk-through :)